### PR TITLE
Support deployment at management group scope

### PR DESCRIPTION
**Pull request by Tiano2017**
## Description

Today we support template deployment at subscription scope and resource group scope. This PR is to add support for management group scope, so now all the *-AzDeployment cmdlets can work on deployments at a management group scope. 

## Checklist

- [ ] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#updating-the-change-log)
- [ ] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/Resources/Resources.Test/Resources.Test.csproj [hunk 1]
```diff
@@ -14,7 +14,6 @@
     <PackageReference Include="Microsoft.Azure.Graph.RBAC" Version="3.5.0-preview" />
     <PackageReference Include="Microsoft.Azure.Management.Authorization" Version="2.11.0-preview" />
     <PackageReference Include="Microsoft.Azure.Management.ManagementGroups" Version="1.1.1-preview" />
-    <PackageReference Include="Microsoft.Azure.Management.ResourceManager" Version="2.1.0-preview" />
```
Comment (Tiano2017) [inline src/Resources/Resources.Test/Resources.Test.csproj]: @cormacpayne do I need to update the ResourceManager package version in TestFx and ScenarioTest.ResourceManager project?

**Comment by cormacpayne** [comment]
@Tiano2017 it looks like there are some failing tests -- would you mind taking a look?

**Comment by Tiano2017** [comment]
Thanks @cormacpayne. I'll take a look at the test failures. I still have more changes to add. Will combine them in my next push.

**Comment by markcowl** [comment]
Closign this PR for inactivity.  Please open a PR when you are ready to release your changes.

**Pull request closed without merging**